Stacks class testing

laurengastineau opened the pull request:
Updated:
* Draggability and droppability tests for `app/cards.js` are now uncommented and are run in `test/card-test.js`
* Added buttons (close, save, fullscreen mode) to card class `app/cards.js`.	

Added:
* `test/stacks.js`  provides basic testing infrastructure for ensuring that a stack instance is set up correctly.  

`Stack tests` check:
* A stack contains the following content:
    * Body div, close button, expand button, annotation text field
*  A stack instance tracks its contained cards.
* A stack instance is collapsed by default.
* Multiple stack instances can be created.
* Multiple stack instances track different sets of cards.
* Cards can be removed from a stack instance.
* Cards are positioned correctly in a stack instance. (This test is currently commented out for now.)

nelsonni commented on test/card-test.js:
```diff
@@ -92,15 +91,15 @@ describe('cards interactions', function () {
-  //   assert.notEqual(card.saveButton, undefined, msg2);
-  //   assert.notEqual(card.fullscreenButton, undefined, msg3);
-  // });
+  it('document contains close, expand, and save buttons', function () {
```
Testing for the existence of elements within an instance of the `Card` class does not test behavior, but instead tests the definition of the `Card` class. This definition could change in the future and still remain a valid `Card`, however, this test would indicate a change in structure is invalid. I suggest removing this test.

nelsonni commented on test/card-test.js:
```diff
@@ -122,16 +121,17 @@ describe('cards interactions', function () {
-  //     var disabled = 1;
-  //   return assert.equal(disabled, 1);
-  // });
+  it('card dragability can be disabled', function () {
```
Testing whether a local variable (`disabled`) equals 1 after evaluating an `if` statement which limits that variable from being declared only when a `Card` class instance allows the `jquery-ui.draggable('disable')` command is a masked testing methodology. Since the ability to disable draggability is inherited from the `jquery-ui` package, this test is actually evaluating the functionality of that package. Tests need to focus on evaluating the functionality of the `synectic` app.

nelsonni commented on test/card-test.js:
```diff
@@ -143,16 +143,17 @@ describe('cards interactions', function () {
-  //     var disabled = 1;
-  //   return assert.equal(disabled, 1);
-  // });
+  it('card dropability can be disabled', function () {
```
Similar to my comments on line 124, the `jquery-ui.droppable('disable')` command is a functionality provided by `jquery-ui` package. No need to test `jquery-ui` within our app; that type of testing should be handled by the developers of `jquery-ui` within their package.

nelsonni commented on test/stacks-test.js:
```diff
@@ -0,0 +1,118 @@
+var Canvas = require('../app/Canvas.js');
+var Stack = require('../app/Stacks.js');
+
+var app = new Application({
```
This declaration of `app` is masked by the declaration of another `app` on line 25. Therefore, this declaration can be removed.

nelsonni commented:
The `stacks-class-testing` branch appears to be slightly behind the `master` branch to which this pull request is attempting to merge onto. Prior to being able to integrate this PR, the following commits need to be pulled into the `stacks-class-testing` branch: f2e68a262cca2f3401466f74916dfcb6cdcc4f8f, 8dd84fde715cf67b061665a428aebf25e4cf5eae, and aabd16cf98caf8ac1d8309a6b958ec0e3c2550d4.

laurengastineau commented:
In my last commit I was mistaken. I have not competed every PR requests for app/card-test.js. I have removed the test that checks for buttons on a card instance, but I am still working on creating a test that checks that a card can be dragged and dropped.
